From 5e16c24920ed920590941ba7a994fc83d2d1b414 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Thu, 30 Jul 2015 09:54:48 +0200
Subject: [PATCH] tests: fix audiomixer test on big endian systems

---
 tests/check/elements/audiomixer.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/check/elements/audiomixer.c b/tests/check/elements/audiomixer.c
index 1d1b091..f30cccd 100644
--- a/tests/check/elements/audiomixer.c
+++ b/tests/check/elements/audiomixer.c
@@ -86,7 +86,11 @@ GST_START_TEST (test_filter_caps)
   GstPad *pad;
 
   filter_caps = gst_caps_new_simple ("audio/x-raw",
+#if G_BYTE_ORDER == G_BIG_ENDIAN
+      "format", G_TYPE_STRING, "F32BE",
+#else
       "format", G_TYPE_STRING, "F32LE",
+#endif
       "layout", G_TYPE_STRING, "interleaved",
       "rate", G_TYPE_INT, 44100, "channels", G_TYPE_INT, 1, NULL);
 
-- 
2.4.3

